Python/traceback.c

```diff
@@ -545,23 +551,32 @@ display_source_line_with_margin(PyObject *f, PyObject *filename, int lineno, int
         *truncation = i - indent;
     }
 
-    if (err == 0) {
-        err = _Py_WriteIndentedMargin(margin_indent, margin, f);
+    assert(!err);
```

Comment:
`err` is set twice: first at initialisation (set to `0`), and then set to `-1` if `linjeobj == NULL` in the get-lineno-loop. If `lineobj == NULL`, we bail after closing `fob` and cleaning up refs. I think it would be an improvement to remove `err` and simply `return -1` in line 523. Then we'd also get rid of this slightly confusing (to me) `assert` :)

Reply:
Very good!